tests/test_java.py

```diff
@@ -691,3 +691,28 @@ def test_meminfo_logged(
     ) as profiler:
         snapshot_pid_profile(profiler, application_pid)
         assert "async-profiler memory usage (in bytes)" in caplog.text
+
+
+# test that java frames include no semicolon but use a pipe '|' character instead, as implemented by AP
+@pytest.mark.parametrize("in_container", [True])
+def test_java_frames_include_no_semicolons(
+    tmp_path: Path,
+    application_pid: int,
+) -> None:
+    with make_java_profiler(
+        storage_dir=str(tmp_path),
+        duration=3,
+        frequency=999,
+    ) as profiler:
+        collapsed = snapshot_pid_profile(profiler, application_pid).stacks
+        # JVM ends object-type signatures with a semicolon; in decompiled output of a class file semicolon appears
+        # after each object type. async-profiler replaces them with '|' to have consistent collapsed stacks format.
+        # We make sure semicolons do not appear neither in function arguments list nor in the return type.
+        # make sure pipe character is used where semicolon would be:
+        assert is_function_in_collapsed("Fibonacci.main([Ljava/lang/String|)V_[j];", collapsed)
+        # make sure semicolon is used only to separate function frames:
+        assert is_function_in_collapsed("Fibonacci$1.run()V_[j];java/io/File.list()[Ljava/lang/String|_[j]", collapsed)
+        # no semicolon in return type:
+        assert not is_function_in_collapsed(";_[j];", collapsed)
+        # no semicolon in arguments list signatures:
+        assert not is_pattern_in_collapsed(r"\([^);]+[;][^)]*\)", collapsed)
```

Comment:
```suggestion
        assert not is_pattern_in_collapsed(r"\([^);]+;[^)]*\)", collapsed)
```
Simpler for the same purpose?

For a stronger check, let's also test the "positive case", i.e
```
assert is_pattern_in_collapsed(r"\([^)|]+[^)]*\)", collapsed)
```
(or whatever - with the `|` instead of `;`, this proves that the regex is correct; because the regex might not match due to other reasons as well)

Reply:
okay, added another pattern.